src/Components/Components/src/RenderTree/RenderTreeFrame.cs

```diff
@@ -17,13 +16,22 @@ namespace Microsoft.AspNetCore.Components.RenderTree
         // Note that the struct layout has to be valid in both 32-bit and 64-bit runtime platforms,
         // which means that all reference-type fields need to take up 8 bytes (except for the last
         // one, which will be sized as either 4 or 8 bytes depending on the runtime platform).
-        // This is not optimal for the Mono-WebAssembly case because that's always 32-bit so the
-        // reference-type fields could be reduced to 4 bytes each. We could use ifdefs to have
-        // different fields offsets for the 32 and 64 bit compile targets, but then we'd have the
-        // complexity of needing different binaries when loaded into Mono-WASM vs desktop.
-        // Eventually we might stop using this shared memory interop altogether (and would have to
-        // if running as a web worker) so for now to keep things simple, treat reference types as
-        // 8 bytes here.
+
+        // Although each frame type uses the slots for different purposes, the runtime does not
+        // allow reference type slots to overlap with each other or with value-type slots.
```

Comment:
> the runtime does not allow reference type slots to overlap with each other or with value-type slots

CoreCLR does allow this, but it produces unverifiable (and thus unsafe) code. I don't recommend we do it, but I have built and thrown away things that do it before 😈

Reply:
OK, well if one day we get desperate to shrink `RenderTreeFrame` by a small amount, we can consider doing that! For now let's keep it safe.